Lib/posixpath.py

```diff
@@ -385,17 +385,37 @@ def normpath(path):
         return _path_normpath(path) or "."
 
 
-def abspath(path):
-    """Return an absolute path."""
-    path = os.fspath(path)
-    if isinstance(path, bytes):
-        if not path.startswith(b'/'):
-            path = join(os.getcwdb(), path)
-    else:
-        if not path.startswith('/'):
-            path = join(os.getcwd(), path)
-    return normpath(path)
-
+try:
+    from posix import _path_abspath
+except ImportError:
+    def abspath(path):
+        """Return an absolute path."""
+        path = os.fspath(path)
+        if isinstance(path, bytes):
+            if not path.startswith(b'/'):
+                path = join(os.getcwdb(), path)
+        else:
+            if not path.startswith('/'):
+                path = join(os.getcwd(), path)
+        return normpath(path)
+else:
+    def abspath(path):
+        """Return an absolute path."""
+        path = os.fspath(path)
+        if isinstance(path, bytes):
+            if path.startswith(b'/'):
+                return os.fsencode(_path_abspath(os.fsdecode(path), 0))
+            else:
+                cwd = os.fsdecode(os.getcwdb())
```

Comment:
Can we use this instead?

```suggestion
                cwd = os.getcwd()
```

Reply:
It should be functionally the same, but a bit faster. The `getcwd()` POSIX system call returns a byte string. If the working directory contains characters that can't be decoded by the filesystem encoding (usually UTF-8), Python's `os.getcwd()` uses the "surrogateescape" error handler. It roundtrips back to the original bytes at the end via `os.fsencode()`.

Reply:
```zsh
wannes@Stefans-iMac cpython % python -m timeit -s "import os" "os.fsdecode(os.getcwdb())" && python -m timeit -s "import os" "os.getcwd()"
20000 loops, best of 5: 14.3 usec per loop # before
20000 loops, best of 5: 13.5 usec per loop # after
# -> 1.06x faster
```